server/service/src/sync/translations/stocktake_line.rs

```diff
@@ -106,6 +107,23 @@ impl SyncTranslation for StocktakeLineTranslation {
             None
         };
 
+        // omSupply should be generating the stocktake line with valid stock lines.
+        // Currently a uuid is assigned by central for the stock_line id which causes a foreign key constraint violation
+        let is_stock_line_valid = match item_line_ID {
+            Some(ref stock_line_id) => StockLineRowRepository::new(connection)
+                .find_one_by_id(stock_line_id)?
+                .is_some(),
+            None => true,
+        };
+
+        if !is_stock_line_valid {
+            log::warn!(
+                "Stock line is not valid, stocktake_line_id: {}, stock_line_id: {:?}",
+                ID,
+                item_line_ID
+            );
+        }
+
```

Comment:
This is just logging... I think need something along the lines of:

```
           let stock_line_id = match is_stock_line_valid {
                true => item_line_ID,
                false => None,
            }
```

Reply:
Already done 🤣